tests/Unit/PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Test_Kuzmin.cpp

```diff
@@ -0,0 +1,126 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <random>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Kuzmin.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+
+void test_create() noexcept {
+  const auto sol =
+      TestHelpers::test_creation<ScalarAdvection::Solutions::Kuzmin>("");
+  CHECK(sol == ScalarAdvection::Solutions::Kuzmin());
+}
+
+void test_serialize() noexcept {
+  ScalarAdvection::Solutions::Kuzmin sol;
+  test_serialization(sol);
+}
+
+void test_move() noexcept {
+  ScalarAdvection::Solutions::Kuzmin sol;
+  ScalarAdvection::Solutions::Kuzmin sol_copy;
+  test_move_semantics(std::move(sol), sol_copy);  //  NOLINT
+}
+
+struct KuzminProxy : ScalarAdvection::Solutions::Kuzmin {
+  using ScalarAdvection::Solutions::Kuzmin::Kuzmin;
+
+  using variables_tags = tmpl::list<ScalarAdvection::Tags::U>;
+
+  template <typename DataType>
+  tuples::tagged_tuple_from_typelist<variables_tags> retrieve_variables(
+      const tnsr::I<DataType, 2>& x, double t) const noexcept {
+    return this->variables(x, t, variables_tags{});
+  }
+};
+
+void verify_solution(const gsl::not_null<std::mt19937*> generator,
+                     const size_t number_of_pts_per_region) {
+  // DataVectors to store values
```

Comment:
maybe add a variable `num_pts = 5 * number_of_pts_per_region` to make the code a bit easier to read?

Reply:
Also added a brief comment :)